Create variables.md

sebajps opened the pull request:
Adding new documentation about existing usefull variables to use in templates

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop developer documentation! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines on how to contribute:
https://devdocs.prestashop-project.org/8/contribute/documentation/how/
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.x / 8.x
| Description?  | Please be specific when describing the PR. What did you add, why did you submit it?
| Fixed ticket? | Fixes #{issue number here} if there is a related issue

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

sebajps committed:
Create variables.md

Adding new documentation about existing usefull variables to use in templates

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+weight: 50
+---
+# Global variables for templates
+Prestashop offers a list of predefined global variables to help you use relevant information in the Front End Smarty templates.
```
```suggestion
PrestaShop offers preset global variables for the front office Smarty templates.
```

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+# Global variables for templates
+Prestashop offers a list of predefined global variables to help you use relevant information in the Front End Smarty templates.
+
+The variables were set in [`classes/FrontController.php`](https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/controller/FrontController.php):
```
```suggestion
The variables are set in [`classes/FrontController.php`](https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/controller/FrontController.php):
```

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+	'static_token' => Tools::getToken(false),
+	'token' => Tools::getToken(),
+	'debug' => _PS_MODE_DEV_,
+];
```
```suggestion
    'cart' => $this->cart_presenter->present($cart),
    'currency' => $this->getTemplateVarCurrency(),
    'customer' => $this->getTemplateVarCustomer(),
    'language' => $this->objectPresenter->present($this->context->language),
    'page' => $this->getTemplateVarPage(),
    'shop' => $this->getTemplateVarShop(),
    'core_js_public_path' => $this->getCoreJsPublicPath(),
    'urls' => $this->getTemplateVarUrls(),
    'configuration' => $this->getTemplateVarConfiguration(),
    'field_required' => $this->context->customer->validateFieldsRequiredDatabase(),
    'breadcrumb' => $this->getBreadcrumb(),
    'link' => $this->context->link,
    'time' => time(),
    'static_token' => Tools::getToken(false),
    'token' => Tools::getToken(),
    'debug' => _PS_MODE_DEV_,
];
```

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+* `{$cart.minimalPurchaseRequired}`: Text for a minimal amount for purchase
+
+### Currency (actual shop currency)
+* `{$currency.id}`: Currency Id in Prestashop database
```
```suggestion
* `{$currency.id}`: Currency ID in PrestaShop database
```

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+## List of variables
+
+### User Cart
+* `{$cart.products}`: List of products contained
```
```suggestion
* `{$cart.products}`: List of products from the current cart
```

kpodemski replied:
Now that I see at it I think it would be better to just mention that there's a `$cart` variable and if you want to know all variables you can do:

```
{$cart|dump}
```

or see the corresponding `Presenter`, `CartPresenter`, the same goes for Customer, etc., otherwise, every time something changes in the presenter, we have to update the doc

sebajps replied:
Totally agree! I will do that change

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+* `{$customer.firstname}`: User Firstname 
+* `{$customer.email}`: User email
+* `{$customer.last_passwd_gen}`: Last date user password were changed 
+* `{$customer.birthday}`: User birthday
```
```suggestion
* `{$customer.lastname}`: Customer last name 
* `{$customer.firstname}`: Customer first name 
* `{$customer.email}`: Customer email
* `{$customer.last_passwd_gen}`: The last date customer password was changed 
* `{$customer.birthday}`: Customer birthday
```

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+* `{$shop.fax}`: Shop fax number
+
+### Urls 
+* `{$urls.base_url}`: Web site url base
```
These are not coming from Presenter so I'd keep it 👍🏻

kpodemski commented on themes/reference/templates/variables.md:
```diff
@@ -0,0 +1,182 @@
+Returns an array containing the description and URL of all the paths that the user has browsed from the home page.
+* `{$breadcrumb}`
+
+### lin
```
```suggestion
### link
```

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update themes/reference/templates/variables.md

Co-authored-by: Krystian Podemski <kpodemski@users.noreply.github.com>

sebajps committed:
Update variables.md

Removing details about the Cart and Languages variables to prevent issues with documentation updates for future versions of PrestaShop.

kpodemski commented:
LGTM :) I'll ping @thomasnares for a review just to confirm if it's in the right section of the page, etc.

thomasnares commented:
thanks @sebajps, can i request a format update ? i liked the table presentation in this page : https://devdocs.prestashop-project.org/8/modules/creation/displaying-content-in-front-office/#list-of-smarty-variables

Could you change your list(s) to tables ? 

I can do it if you prefer. 

I will then link https://devdocs.prestashop-project.org/8/modules/creation/displaying-content-in-front-office/#list-of-smarty-variables to this page (and drop its content about smarty variables), as your list is more complete.

sebajps commented:
Hi Thomas, thanks! If you can help me with the tables would be great :).
I have doubts about the path where it will be located. For example, this
doc will also usefull for Themes section for working with templates:
https://devdocs.prestashop-project.org/8/themes/reference/templates/

What do you think?
Thanks!
 Sebastian


El mar, 16 may 2023 a las 10:09, Thomas NARES ***@***.***>)
escribió:

> thanks @sebajps <https://github.com/sebajps>, can i request a format
> update ? i liked the table presentation in this page :
> https://devdocs.prestashop-project.org/8/modules/creation/displaying-content-in-front-office/#list-of-smarty-variables
>
> Could you change your list(s) to tables ?
>
> I can do it if you prefer.
>
> I will then link
> https://devdocs.prestashop-project.org/8/modules/creation/displaying-content-in-front-office/#list-of-smarty-variables
> to this page (and drop its content about smarty variables), as your list is
> more complete.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PrestaShop/docs/pull/1656#issuecomment-1549198716>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5Z5PTHQUFGOIPBF5MP32LLXGMY2BANCNFSM6AAAAAAX4Q3I6I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Commit:
Change markdown, add link on other page

thomasnares commented:
@sebajps, i opened a PR on your branch : https://github.com/sebajps/docs/pull/1

Thanks 👍

thomasnares committed:
Update themes/reference/templates/variables.md

sebajps committed:
Merge pull request #1 from thomasnares/improveVariables

Change markdown, add link on other page

kpodemski committed:
fix typo

kpodemski commented:
thank you @sebajps @thomasnares :)

sebajps commented:
Guys in the modules section keeps as before, right?
The header title for smarty variables says “file/folder”, I think should be
“variable”

https://devdocs.prestashop-project.org/8/modules/creation/displaying-content-in-front-office/


El El lun, 22 may 2023 a las 10:30, Krystian Podemski <
***@***.***> escribió:

> thank you @sebajps <https://github.com/sebajps> @thomasnares
> <https://github.com/thomasnares> :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PrestaShop/docs/pull/1656#issuecomment-1556775103>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5Z5PTGHTYVNBZ36UUQXYXLXHMPYZANCNFSM6AAAAAAX4Q3I6I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
[image: prestashop monogram]
Sebastián Jaimovich <https://www.linkedin.com/in/sebaj/>
Expert Solution Engineer - Spain & LATAM
[image: prestashop logo]
[image: facebook] <https://www.facebook.com/prestashop> [image: twitter]
<https://twitter.com/PrestaShop> [image: youtube]
<https://www.youtube.com/prestashop> [image: instagram]
<https://www.instagram.com/prestashop/> [image: linkedin]
<https://www.linkedin.com/company/prestashop>

kpodemski commented:
@sebajps your changes are not live yet, CD is still running

sebajps commented:
Ok, thanks but I think the title I sent you in not ok or that page will be
removed?

El El lun, 22 may 2023 a las 10:37, Krystian Podemski <
***@***.***> escribió:

> @sebajps <https://github.com/sebajps> your changes are not live yet, CD
> is still running
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PrestaShop/docs/pull/1656#issuecomment-1556785736>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5Z5PTH3KBCIKK6VKTWDBDDXHMQTRANCNFSM6AAAAAAX4Q3I6I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
[image: prestashop monogram]
Sebastián Jaimovich <https://www.linkedin.com/in/sebaj/>
Expert Solution Engineer - Spain & LATAM
[image: prestashop logo]
[image: facebook] <https://www.facebook.com/prestashop> [image: twitter]
<https://twitter.com/PrestaShop> [image: youtube]
<https://www.youtube.com/prestashop> [image: instagram]
<https://www.instagram.com/prestashop/> [image: linkedin]
<https://www.linkedin.com/company/prestashop>

kpodemski commented:
I checked with @sebajps on Slack, and everything works ok 👍🏻
